src/main/java/dk/kb/datahandler/oai/OaiResponseFilterPreservica.java

```diff
@@ -81,10 +81,12 @@ public void addToStorage(OaiResponse response) throws ApiException {
             if (oaiRecord.getId().contains("oai:col")){
                 continue;
             }
-            // DeliverableUnits need to have the PBCore metadata tag. Manifestations does not seem to have it.
+            // DeliverableUnits from preservica 5 and InformationObjects from preservcia 6 need to have the PBCore metadata tag.
+            // Manifestations from preservica 5 does not seem to have it.
             // Therefore, we are checking the ID as well.
             Matcher metadataMatcher = METADATA_PATTERN.matcher(xml);
-            if (oaiRecord.getId().contains("oai:du") && !metadataMatcher.find()){
+            if (oaiRecord.getId().contains("oai:du") || oaiRecord.getId().contains("oai:io")
+                    && !metadataMatcher.find()){
```

Comment:
[Operator precedence](https://docs.oracle.com/javase/tutorial/java/nutsandbolts/operators.html) says `&&` is evaluated before `||` so the expression is equivalent to 
```Java
oaiRecord.getId().contains("oai:du") || 
(oaiRecord.getId().contains("oai:io") && !metadataMatcher.find())
```
where it should have been
```Java
(oaiRecord.getId().contains("oai:du") || oaiRecord.getId().contains("oai:io")) &&
!metadataMatcher.find())
```

Reply:
Oh, I didn't know about the precedence. Good to know!